app/src/main.js

```diff
@@ -27,14 +27,22 @@ var vm = new Vue({
 });
 
 vm.$on("subregionSelected", function(d) {
+  if (d === "All") {
+    $(".map").css({ fill: "" });
+    this.$children[0].$children[4].$children[0].selectedRegion = {};
+    this.$children[0].$children[4].selectedRegion = {};
+    this.$root.showMain = true;
+    this.$root.selectedSubregion = false;
```

Comment:
You don't need to do lines 34 and 35. Instead of 35, you should do `selectedSubregion.update({})`